gamla/io_utils.py

```diff
@@ -272,21 +272,42 @@ async def post_json_with_extra_headers_and_params_async(
 
 
 @currying.curry
-def retry(
+def retry_with_count(
     exception: Union[Exception, Tuple[Exception, ...]],
+    max_times: int,
```

Comment:
Every time you would call this function you will give `times` and `max_times` as the same number. 
To avoid this just have an inner or helper private function.